pkg/receive/handler.go

```diff
@@ -74,48 +65,69 @@ func instrumentHandler(handlerName string, handler http.HandlerFunc) http.Handle
 	)
 }
 
+func NewHandlerInMux(mux *http.ServeMux, logger log.Logger, o *Options) *Handler {
+	h := NewHandler(logger, o)
+	mux.Handle("/", h.router)
+	return h
+}
+
 func NewHandler(logger log.Logger, o *Options) *Handler {
 	router := route.New().WithInstrumentation(instrumentHandler)
 	if logger == nil {
 		logger = log.NewNopLogger()
 	}
 
 	h := &Handler{
-		logger:       logger,
-		router:       router,
-		readyStorage: o.ReadyStorage,
-		receiver:     o.Receiver,
-		options:      o,
-		quitCh:       make(chan struct{}),
+		logger:          logger,
+		router:          router,
+		readyStorage:    o.ReadyStorage,
+		receiver:        o.Receiver,
+		options:         o,
+		quitCh:          make(chan struct{}),
+		readinessProber: o.ReadinessProber,
 	}
 
 	readyf := h.testReady
 	router.Post("/api/v1/receive", readyf(h.receive))
+	o.Registry.MustRegister(
+		requestDuration,
```

Comment:
Where is this coming from? Cannot see new metrics in your PR?

Reply:
Yes.. those were added by @brancz in the initial PR with the receive but were never registered. 
I took a stab here since I was touching the code anyway but yes, it's out of scope of this PR. 
Should I remove it?

Reply:
Nah all good, leave it here.